pymongo/collection.py

```diff
@@ -289,6 +289,10 @@ def __ne__(self, other):
     def __hash__(self):
         return hash((self.__database, self.__name))
 
+    def __bool__(self):
+        raise NotImplementedError("Collection does not implement Boolean "
```

Comment:
This error message here should be more helpful. Try to put yourself in the user's shoes. Suppose you had an app that was relying on this feature:
```python
if some_condition():
    coll = client.test.test
else:
    coll = None

if coll:  # This line worked in Pymongo 3 but will raise an error in PyMongo 4
    coll.find_one()
```

What would you want the error message to say? It should be something that makes it clear what the developer should do to resolve the error.

Reply:
I tried to improve the error message.

Reply:
Great, it's much better now! I suggest a few tweaks to better match the language used in [Python's documentation for `__bool__`](https://docs.python.org/3/reference/datamodel.html#object.__bool__):
```
Collection objects do not implement truth value testing or bool(). Please compare with None instead: collection is not None
```

Reply:
Done.